lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1904,6 +1904,17 @@ ParseResult FIRStmtParser::parseMemPort(MemDirAttr direction) {
   return addSymbolEntry(id, result, info.getFIRLoc());
 }
 
+/// Returns true if \p base begins with \p s. This shold be a special cased
+/// implementation of std::string::find, where the string match is performed
+/// only at the beggining of the base. Should be more efficient than the find.
+static bool stringBeginsWith(const std::string &base, const std::string &s) {
+  for (unsigned i = 0, e = s.size(); i < e; ++i) {
+    if (base[i] != s[i])
+      return false;
+  }
+  return true;
+}
+
```

Comment:
Any reason to not use [`llvm::StringRef::startsWith(StringRef prefix)`](https://llvm.org/doxygen/classllvm_1_1StringRef.html#aeb540aeadd339af860f5c217a85363b7)?

Reply:
Thanks, missed checking for StringRef. Now using this.